### PR TITLE
Prevent rendering portals on elments that do not exist

### LINKED ISSUE #3568: Using Plain Permalinks Breaks the Entry Editor
## Replication steps

1. Change Permalink settings to plain ( ?p=123 )
2. Install Caldera Forms
3. Make a form
4. Submit the form
5. View entry
6. Edit entry
7. See that the entry editor is covered by a home page

## Debug Info
WordPress Version: 5.4.2

PHP Version: 7.3.19

MySQL Version: 5.7.30

Server Software: Apache/2.4.38 (Debian)

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36

Session Save Path:

Session Save Path Exists: No

Session Save Path Writeable: No

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
Redis: No
Object Cache:

Apc: No
Apcu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 40M

Currently Active Theme: Twenty Nineteen: 1.5

Plugins:

Akismet Anti-Spam: 4.1.6 Not Active
Authorize.net for Caldera Forms: 2.1.4 Active
Caldera Forms: 1.9.1 Active
CalderaWP License Manager: 1.2.10 Active
MailPoet 2: 2.13 Not Active
Ninja Forms: 3.4.24.3 Not Active
Ninja Forms - ActiveCampaign: 3.0.6 Not Active
Ninja Forms - Addon Manager: 3.0.13 Not Active
Ninja Forms - AWeber: 3.1.1 Not Active
Ninja Forms - Campaign Monitor: 3.0.5 Not Active
Ninja Forms - Capsule CRM: 3.4.0 Not Active
Ninja Forms - CleverReach: 3.1.3 Not Active
Ninja Forms - ClickSend: 3.0.1 Not Active
Ninja Forms - Conditional Logic: 3.0.26 Not Active
Ninja Forms - Constant Contact: 3.0.4 Not Active
Ninja Forms - ConvertKit: 3.0.2 Not Active
Ninja Forms - Elavon Payment Gateway: 3.1.0 Not Active
Ninja Forms - EmailOctopus: 3.0.0 Not Active
Ninja Forms - Emma: 3.0.4 Not Active
Ninja Forms - Excel Export: 3.3.1 Not Active
Ninja Forms - File Uploads: 3.3.4 Not Active
Ninja Forms - Help Scout: 3.1.3 Not Active
Ninja Forms - Highrise CRM: 3.0.0 Not Active
Ninja Forms - Insightly CRM: 3.2.0 Not Active
Ninja Forms - Layout & Styles: 3.0.28 Not Active
Ninja Forms - Mail Chimp: 3.1.11 Not Active
Ninja Forms - MailPoet: 3.0.0 Not Active
Ninja Forms - Multi-Part Forms: 3.0.26 Not Active
Ninja Forms - OnepageCRM: 3.0.0 Not Active
Ninja Forms - PayPal Express: 3.0.15 Not Active
Ninja Forms - PDF Form Submissions: 3.1.2 Not Active
Ninja Forms - Pipeline Deals CRM: 3.0.1 Not Active
Ninja Forms - Post Creation: 3.0.9 Not Active
Ninja Forms - Recurly: 3.0.4 Not Active
Ninja Forms - Salesforce CRM: 3.2.0 Not Active
Ninja Forms - Save Progress: 3.0.24.2 Not Active
Ninja Forms - Slack: 3.0.3 Not Active
Ninja Forms - Stripe: 3.1.3 Not Active
Ninja Forms - Trello: 3.0.3 Not Active
Ninja Forms - Twilio SMS: 3.0.1 Not Active
Ninja Forms - User Analytics: 3.0.0 Not Active
Ninja Forms - User Management: 3.0.12 Not Active
Ninja Forms - Webhooks: 3.0.5 Not Active
Ninja Forms - WebMerge: 3.0.3 Not Active
Ninja Forms - Zapier: 3.0.8 Not Active
Ninja Forms - Zoho CRM: 3.4 Not Active
Slack Integration for Caldera Forms: 1.0.0 Active
Verify Email for Caldera Forms: 1.2.0 Not Active
WP Mail Logging: 1.9.6 Not Active

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #3568

Fixes https://github.com/CalderaWP/cf-connected-forms/issues/76

## What Has Changed
Updated form-builder library so we can have this commit: https://git.saturdaydrive.io/_/caldera-forms/tools/form-builder/commit/4458d6f53f0c9943ca398705ec9e51d5425f2a10

In connected forms, many portals we render on normally do not exist. This updates the portal render function to not render if domNode is null.

## How To Test
Check out this branch and use this connecte forms zip

https://github.com/CalderaWP/cf-connected-forms/blob/master/releases/cf-connected-forms-1.3.0-b-2.zip

* Check that save button, processor conditionals, conditionals all work in normal form editor.
* Check that the connected forms bug is resolved

### CODE REVIEW

**Comment by New0** [comment]
@Shelob9 I tested this branch after refreshing the assets 

- Regular Forms editore tests worked as expected, conditions and saving

- The Save button on the Connected Forms editor appeared but clicking on it resulted in an error for   onClick index.min.js:2518
    
```
Uncaught TypeError: can't assign to property "conditions" on true: not an object
    onClick https://local.lndo.site/wp-content/plugins/caldera-forms/clients/form-builder/build/index.min.js?ver=1.9.1:1
    onClick https://local.lndo.site/wp-content/plugins/caldera-forms/clients/form-builder/build/index.min.js?ver=1.9.1:1
    Vh https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:163
    Uh https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:14
    Xh https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:14
    af https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:14
    Yh https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:164
    nd https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:15
    nc https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:15
    Of https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:38
    Ac https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:39
    unstable_runWithPriority https://local.lndo.site/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26
    Ma https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    Be https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:119
    xi https://local.lndo.site/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:39
```

Line 2518 leads to this portion of the code where the "type" seems to return the error
```
e.config.hasOwnProperty('fields') && Object.keys(e.config.fields).forEach((function (n) {
              e.config.fields.hasOwnProperty(n) && (e.config.fields[n].conditions = {
                type: ''
              })
})),

```
Nothing gets logged in debug.log file and not request is triggered yet in the Network inspector

**Comment by Shelob9** [comment]
@New0 it makes sense that the conditions and fields are not set, this:
https://github.com/CalderaWP/Caldera-Forms/blob/master/clients/form-builder/index.js#L104-L150

Assumes a form, not a connected forms.

**Comment by New0** [comment]
@Shelob9 I changed the condition to check for the definition of `fieldID`
When undefined, an empty array return true which resulted in the error " can't assign to property "conditions" on true: not an object ". 
And `conditions` variable also returns an empty array when there aren't conditions

**Comment by New0** [comment]
@Shelob9 I built a beta 3, that could help to have @riverBanjo testing that full fixes with Connected forms included.
[1.9.2-beta3.zip](https://github.com/CalderaWP/Caldera-Forms/files/4930685/1.9.2-beta3.zip)

File: clients/form-builder/index.js [hunk 1]
```diff
@@ -124,19 +124,21 @@ const HandleSave = ({ jQuery, formId }) => {
...
-		if (data_fields.config.hasOwnProperty("fields")) {
-			Object.keys(data_fields.config.fields).forEach((fieldId) => {
-				if (data_fields.config.fields.hasOwnProperty(fieldId)) {
-					data_fields.config.fields[fieldId].conditions = {
-						type: "",
-					};
-				}
-			});
-		}
+		if( typeof fieldID !== "undefined" ){
```
Comment (Shelob9) [inline clients/form-builder/index.js:135]: @New0 I don't see `fieldID` defined in the scope of HandleSave function.

Comment (New0) [reply clients/form-builder/index.js:135]: @Shelob9 you're right, I'm so confused. I am able to save Form settings with conditionals using that condition and it also fixes the issue of the connected forms... I might have misread the l36 thinking it was globally defining a fieldID.

Comment (Shelob9) [reply clients/form-builder/index.js:135]: I see what you mean. While I was looking at L36 I saw:  `const { formFields } = React.useContext(FormFieldsContext);` Maybe we could do that in `handleSave` and check if formFields is undefined or is empty array -- what it should be.

**Comment by New0** [comment]
@Shelob9 I checked the length of conditions instead, I think that makes more sense... My manual tests pass, I was able to save Regular and Connected forms and save conditions as expected.

Here is the new beta: 
[1.9.2-beta4.zip](https://github.com/CalderaWP/Caldera-Forms/files/4938838/1.9.2-beta4.zip)

**Comment by Shelob9** [comment]
Make sense to me. I approve of this change now.

**Pull request merged**